Makefile

```diff
@@ -83,26 +83,22 @@ bin/Test$(CONFIGURATION)/$(NATIVE_TIMING_LIB): tests/NativeTiming/timing.c $(wil
 	mkdir -p `dirname "$@"`
 	gcc -g -shared -o $@ $< -m32 $(JI_JDK_INCLUDE_PATHS:%=-I%)
 
-bin/Test$(CONFIGURATION)/Java.Interop-Tests.dll: $(wildcard src/Java.Interop/*/*.cs src/Java.Interop/Tests/*/*.cs)
+# Usage: $(call TestAssembly,assembly-basename)
+define TestAssembly
+bin/Test$(CONFIGURATION)/$(1)-Tests.dll: $(wildcard src/$(1)/*/*.cs src/$(1)/Test*/*/*.cs)
```

Comment:
IIUC, this will expand `$(CONFIGURATION)`, `$(XBUILD)` etc at the time of call itself, which in a sense hardcodes the values of those in the rule. Should `$$(CONFIGURATION)` be used instead?